### DISCUSSION TITLE
Use Anon Records instead of tuples in AST?

### DISCUSSION THREAD

**Discussion by Szer** [dotnet/fsharp#11396] [Ideas]
You know the pain.

Now
```fsharp
| SynPat of (a,b,c,d,e,f,g,...)
...
let (SynPat(_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,_,x,_,) = synPat
```

What it could be
```fsharp
| SynPat of {| Ident: Ident; Range: Range; Attributes: AttrList; ...)
...
let (SynPat {| AttrList = x }|) = synPat
```

Pros
- No more endless discarding, just get what you need
- Named access to the fields (impossible in tuples)

Cons
- Breaking consumers one more time (we got used to it anyway)

**Comment by Szer** [comment]
Okay, I've realized that Anon Records doesn't work in pattern matching at all, but it was a good try :D

**Reply by auduchinok** [reply]
Moreover, they don't work outside the project they are used in.

**Comment by baronfel** [comment]
Per the [docs](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/discriminated-unions) you can actually match on only specific named fields of a DU, so that pain point you listed is taken care of.
